packages/contracts-bedrock/src/Safe/VetoerSet/OwnerGuard.sol

```diff
@@ -0,0 +1,134 @@
+
+    /// @notice The initial max owner count used at deployment.
+    uint8 public constant INITIAL_MAX_OWNER_COUNT = 7;
+
+    /// @notice The Safe Account for which this contract will be the guard.
+    Safe public immutable safe;
+
+    /// @notice The maximum number of owners. Can be changed by supermajority.
+    uint8 public maxOwnerCount;
+
+    /// @notice Thrown at deployment if the current Safe Account owner count can't fit in a `uint8`.
+    /// @param ownerCount The current owner count.
+    error InvalidOwnerCount(uint256 ownerCount);
+
+    /// @notice Thrown if the new owner count is above the `maxOwnerCount` limit.
+    /// @param ownerCount The Safe Account owner count.
+    /// @param maxOwnerCount The current `maxOwnerCount`.
+    error OwnerCountTooHigh(uint256 ownerCount, uint256 maxOwnerCount);
+
+    /// @notice Thrown after the Safe Account executed a transaction if its threshold does not matches
+    ///         with the desired 66% threshold.
+    /// @param threshold The Safe Account threshold.
+    /// @param expectedThreshold The expected threshold.
+    error InvalidSafeAccountThreshold(uint256 threshold, uint256 expectedThreshold);
+
+    /// @notice Thrown when trying to update the `maxOwnerCount` limit but the caller is not
+    ///         the associated Safe Account.
+    /// @param sender The sender address.
+    error SenderIsNotSafeAccount(address sender);
+
+    /// @notice Thrown when trying to update the `maxOwnerCount` limit to a value lower than the current
+    ///         Safe Account owner count.
+    /// @param newMaxOwnerCount The new invalid max count.
+    /// @param ownerCount The current Safe Account owner count.
+    error MaxOwnerCountTooLow(uint256 newMaxOwnerCount, uint256 ownerCount);
+
+    /// @notice Constructor.
+    /// @param _safe The Safe Account for which this contract will be the guard.
+    constructor(Safe _safe) {
+        safe = _safe;
+
+        // Ensure the current number owners of the Safe Account can fit in a `uint8`.
+        uint256 ownerCount = _safe.getOwners().length;
+        if (ownerCount > type(uint8).max) {
+            revert InvalidOwnerCount(ownerCount);
+        }
+
+        // Set the initial `maxOwnerCount`, to the greater between `INITIAL_MAX_OWNER_COUNT` and the current owner
+        // count.
+        maxOwnerCount = uint8(FixedPointMathLib.max(INITIAL_MAX_OWNER_COUNT, ownerCount));
+    }
+
+    /// @notice Inherited hook from `BaseGuard` that is run right before the transaction is executed
+    ///         by the Safe Account when `execTransaction` is called.
+    function checkTransaction(
+        address,
+        uint256,
+        bytes memory,
+        Enum.Operation,
+        uint256,
+        uint256,
+        uint256,
+        address,
+        address payable,
+        bytes memory,
+        address
+    )
+        external
+    { }
+
+    /// @notice Inherited hook from `BaseGuard` that is run right after the transaction has been executed
```

Comment:
confused that comment says `inherited` but there is no `override` ?

Reply:
Yes maybe unclear, this is because it is inherited from `interface Guard` and the override keyword is not required when overriding an interface function.

Reply:
Got it! Would have expected `IGuard`

Reply:
might be nice to add to natspec what we're checking here and revert cases.